quiz/models.py

```diff
@@ -15,8 +15,8 @@ class Question(models.Model):
     RESULT_CHOICES = (
         ('OK', 'The program is guaranteed to output:'),
         ('CE', 'The program has a compilation error'),
+        ('UD', 'The program has / risks undefined behavior'),
```

Comment:
Other ideas:
- The program has potential undefined behavior
- The program has the potential for undefined behavior
- The program (potentially) has undefined behavior
- The program could encounter undefined behavior
- The program might exhibit undefined behavior

It would be nice if we could say something here that works both for the programs that are guaranteed to encounter UB and the ones that only will do that on some implementations.

Reply:
> say something here that works both for the programs that are guaranteed to encounter UB and the ones that only will do that on some implementations

I like the general idea of rewording. However, it's not easy to pick a good alternative that sounds natural for all cases. For example, I don't like the answers "has potential UB" or "might exhibit UB" for "static" UB cases (e.g., IFNDR for using reserved identifiers), because they sound like there's a possibility to not encounter one in such cases. In the same time, those answers naturally cover "dynamic" cases (e.g., `if (rand() and !ptr) *ptr;`), what we may or may not want. Related links: [1](https://stackoverflow.com/q/23153445/6944845), [2](https://stackoverflow.com/q/18385020/6944845).

Maybe something more verbose like "The program has undefined behavior (on at least one conforming implementation)"?

(We also should not forget to update [this doc](https://github.com/knatten/cppquiz23/blob/master/METADATA_HOWTO.md) afterwards.)

Reply:
Yeah, I agree, that's why I had the "has / risks" and "(potentially) in parentheses" in some of the suggestions. I don't want it to be too long either, though. What about "The program has (or might have) undefined behavior"?

Reply:
True, it's too long to display nicely.

![image](https://github.com/knatten/cppquiz/assets/22129543/1e09e2bc-84c3-4ddb-8b3b-976b4ecdb162)

I like the last suggestion. Maybe even shorter "The program is (or may be) undefined"? It's also more consistent with the US answer.

![image](https://github.com/knatten/cppquiz/assets/22129543/d6bce969-23c2-477f-ac4a-cd6cd557f372)
![image](https://github.com/knatten/cppquiz/assets/22129543/fa0c119d-0752-484a-95e8-96ab84e90636)

Reply:
Yeah let's go with "is (or may be) undefined"!